clang/tools/3c/README.md

```diff
@@ -0,0 +1,65 @@
+
+You can ignore the errors about a compilation database not being found.  You can specify a single set of flags to use for all files by prefixing them with `-extra-arg-before=`, for example:
+
+```
+3c -alltypes -output-postfix=checked -extra-arg-before=-Isome/include/path foo.c bar.c
+```
+
+(If you were using a compilation database, such "extra" flags would be added to any flags in the database.)
+
+The `-alltypes` option causes `3c` to try to infer array types.  We want to make this the default but haven't done so yet because it breaks some things.
+
+## Annotated system headers
+
+As mentioned above, if a Checked C program uses an external library, the library's headers must have Checked C annotations on the declarations of the elements (functions, variables, types, etc.) used by the program in order for the program to access those items safely.  Elements with unannotated declarations can only be accessed unsafely.
+
+In the future, we may have some scheme analogous to [DefinitelyTyped](https://definitelytyped.org/) to distribute Checked C headers for existing C libraries.  In the meantime, the Checked C project maintains annotated versions of the most common "system" header files (`stdio.h`, etc.).  The annotated header files do not include all elements, but they `#include` your system's original header files, so your program can still use unannotated elements unsafely.  If you followed the [build instructions](../../docs/checkedc/3C/README.md), these header files should be present in `llvm/projects/checkedc-wrapper/checkedc/include`.
+
+Currently, the annotated header files are named with a `_checked.h` suffix, e.g., `stdio_checked.h`, so that `stdio_checked.h` can `#include <stdio.h>` without causing infinite recursion.  We hope to switch to `#include_next` and remove the suffix soon.  In the meantime, you have to modify your code to `#include <stdio_checked.h>` instead of `stdio.h` and so forth.  The `clang/tools/3c/utils/update-includes.py` tool will do this for you.  It takes one argument: the name of a file containing a list of paths of `.c` and `.h` files to be updated.  In the example of the previous section, you would run:
+
+```
+cat >list <<EOL
+foo.c
+bar.c
+EOL
+PATH/TO/clang/tools/3c/utils/update-includes.py list
+```
+
+## Flags
+
+An incomplete list of useful flags accepted by `3c`:
+
+- `-warn-root-cause`: Show information about the _root causes_ that prevent `3c` from converting unsafe pointers (`T *`) to safe ones (`_Ptr<T>`, etc.).
+
+(TODO: Did I miss anything?)
```

Comment:
@jackastner Can you enumerate other options that are worth mentioning?

Reply:
There are some in the "Standalone tool" section of the 3C doc (not posting the capability-granting link here).  Mike, do you know which are worth mentioning?

Reply:
`-debug-solver` and `-addcr` are worth mentioning.

Reply:
Can you help me come up with an adequate explanation of these options for public consumption (in Slack if it's easier)?  In particular, does it make sense to list `-debug-solver` without providing any information about how to interpret the output?

Reply:
Not changing anything here for the Microsoft PR.